library/beaker_provisioner.py

```diff
@@ -0,0 +1,419 @@
+
+DOCUMENTATION = '''
+---
+module: beaker_provisioner
+version_added: "0.1"
+short_description: Provision servers via Beaker
+description:
+   - Provision servers via Beaker
+options:
+    server_url:
+        description:
+            - Base URL of Beaker server
+        required: tru
+    username:
+        description:
+            - Login username to authenticate to Beaker
+        required: false
+        default: admin
+    password:
+        description:
+            - Password of login user
+            required: true
+    fqdn:
+        description:
+            - Fully qualified domain name of a system
+        required: true
+    action:
+        description:
+            - Action to perform
+        required: true
+        choices: 'provision', 'release'
+    distro_tree_id:
+        description:
+            - Distro Tree ID
+        default: 71576
+        required: false
+'''
+
+LOAN_COMMENT = 'Loaned by BeakerManager'
+SYS_FQDN_PATH = '/systems/{fqdn}/'
+WAIT_BETWEEN_PROVISION_CHECKS = 10
+
+
+def is_fail(status_code, success_code=200):
+    """
+    Check if a command execution failed or not
+
+    :param status_code: The command status code
+    :param success_code: Success code (default is 200)
+    :return: False if status code is equal to success code, otherwise, False
+    """
+
+    return status_code != success_code
+
+
+class BeakerManager(object):
+    REQ_URL = dict(
+        ALL='',
+        FREE='free/',
+        LOAN='loans/',
+        VIEW='view/',
+        RETURN='loans/+current',
+        RESERVE='reservations/',
+        RELEASE='reservations/+current',
+        PROVISION='installations/'
+    )
+
+    def __init__(self, url, username, password, fqdn,
+                 verify=False, disable_warnings=True):
+        """
+        BeakerManager initializer.
+
+        :param url: Base URL of the Beaker server
+        :param username: Authentication username
+        :param password: Authentication password
+        :param fqdn: Fully qualified domain name of a system
+        :param verify: Whether to Check SSL certificate or not (Can specify a
+        path to a certificates file. Default = False)
+        :param disable_warnings: Hide unverified HTTPS requests warnings (
+               default = False)
+        """
+
+        self.base_url = url
+        self.auth = requests.auth.HTTPBasicAuth(username, password)
+        self.fqdn = fqdn
+        self.changed = False
+
+        self.session = requests.Session()
+        self.session.auth = self.auth
+
+        # verify the SSL certificate or not (and disable warning if needed)
+        self.session.verify = verify
+        if disable_warnings:
+            requests.packages.urllib3.disable_warnings()
+
+        # cookie workaround - consider changing with _get_last_activity()
+        self.get(limit=1)
+
+    def view(self):
+        headers = {'Content-Type': 'application/json'}
+        old_params, new_params = self.session.params, {'tg_format': 'turtle'}
```

Comment:
Backing up the session's params and settings new ones.
Defines the system information data to be in 'turtle' format. (I did it long time ago, and as far as I remember, it was easier to get the info out of this format rather than from HTML)